app/src/main/java/com/udacity/asteroidradar/main/AsteroidAdapter.kt

```diff
@@ -0,0 +1,46 @@
+package com.udacity.asteroidradar.main
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.udacity.asteroidradar.Asteroid
+import com.udacity.asteroidradar.R
+
+class AsteroidAdapter(val onItemClicked: (Asteroid) -> Unit): RecyclerView.Adapter<AsteroidAdapter.Viewholder>() {
+
+    private var asteroids: List<Asteroid> = ArrayList()
+
+    class Viewholder(itemView: View) : RecyclerView.ViewHolder(itemView) {
```

Comment:
Consider renaming this to `ViewHolder`, your h is lowercased

Reply:
You may also want to rename it to something more specific like `AsteroidsViewHolder` so you don't confuse it with the base `ViewHolder` given to you by Android.